core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -47,18 +66,23 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
     }
 }
 
+// is File a could type to use here?
+// No way to get script name if file is not present, should add another property (val fileName: String)
 interface ScriptContents {
     val file: File?
     val annotations: Iterable<Annotation>
```

Comment:
I'm still confused by the fact that `Sequence` is not `Iterable`, but otherwise `Sequence` is a good variant for me.

About `annotations` - the fact that we're able to use only file annotations here is an unfortunate limitation of the current parser, that I believe could be lifted later. So I don't think that this limitation should be reflected in the API.